`glide update` vendors current repo if it contains subpackages

Trying to setup `glide` for first time on an application target that contains subpackages. The problem I'm running into is that it vendors the current repo subpackages since they're imported from the main package. That means `go build` uses the vendor directory rather than the primary source which may contain local edits.

It's a private repo but the glide project has the same setup and repros the issue. 

```
~/code/go/src/github.com/masterminds/glide ❯ glide update
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Masterminds/semver.
[INFO] Fetching updates for gopkg.in/yaml.v2.
[INFO] Fetching updates for github.com/Masterminds/vcs.
[INFO] Fetching updates for github.com/codegangsta/cli.
[INFO] Detected semantic version. Setting version for github.com/Masterminds/semver to 1.0.1.
[INFO] Detected semantic version. Setting version for github.com/Masterminds/vcs to 1.4.0.
[INFO] Resolving imports
[INFO] Not found in vendor/: github.com/Masterminds/glide/action (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/msg (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/path (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/repo (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/util (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/cfg (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/dependency (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/gb (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/godep (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/gpm (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/tree (1)
[INFO] Not found in vendor/: github.com/Masterminds/glide/importer (1)
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Masterminds/glide.
[INFO] Setting references for remaining imports
[INFO] Project relies on 5 dependencies.
```

Is there some other way I should be bootstrapping my project so that `glide` doesn't try to vendor these subpackages? In the glide example those subpackages aren't in the `glide.lock` file but a `glide update` adds them.

I can confirm I have the same or a similar issue to the original poster.
I am running:
Windows 7, updated to latest service pack, etc.
Golang 1.6.
Glide version 9.3 pulled right from master today (2014-03-14, commit d5fc6b676a752b37f2646d91e9a463c97eb00519).

I have no symlinks at all, and GOPATH is set properly.

`glide update` fails to update my `glide.lock` file with the new commit hashes.
If I update them manually and then `glide install` it will work fine.  And if I `glide update` inside a docker container that has golang and glide installed, that also works just fine.

I run my commands, such as `glide update` through either `git-bash` or `mingw`.  I don't seem to have these issues if I run them through the horrible `cmd`.  I have a feeling this is a Windows issue.

Example output below.  In this case `api/ciolite` is my own code, and glide should not be trying to update it.  It also fails to update the version of logrus in the lock file, though it doesn't say any errors about it below.
```
$ glide update
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Sirupsen/logrus.
...
[INFO] Resolving imports
[INFO] Fetching api\ciolite into c:\Users\xxxxxx\workspace\xxxxxx\golang\src\vendor
[ERROR] Error looking for api\ciolite: Cannot detect VCS
...
[WARN] Unable to checkout api/ciolite
[ERROR] Update failed for api/ciolite: Cannot detect VCS
...
Cannot detect VCS
```

My yaml looks approximately like this, but with more packages (all of which start with `github.com`):
```
package: .
import:
- package: github.com/Sirupsen/logrus
```

I can also confirm that I got the error today on linux ubuntu, when specifying the yaml file:
```
glide --home ./src/glide.yaml install
[INFO] Lock file (glide.lock) does not exist. Performing update.
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Sirupsen/logrus.
...
[INFO] Resolving imports
[INFO] Fetching api/ciolite into /xxxxxx/xxxxxx/vendor
[ERROR] Error looking for api/ciolite: Cannot detect VCS
...
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for api/ciolite.
...
[WARN] Unable to checkout api/ciolite
[ERROR] Update failed for api/ciolite: Cannot detect VCS
...
Cannot detect VCS
```

--- a/dependency/resolver.go
+++ b/dependency/resolver.go
@@ -2,6 +2,7 @@ package dependency
 
 import (
 	"container/list"
+	"runtime"
 	//"go/build"
 	"os"
 	"path/filepath"
@@ -216,6 +217,14 @@ func (r *Resolver) Resolve(pkg, basepath string) ([]string, error) {
 	return r.resolveImports(l)
 }
 
+// dirHasPrefix tests whether the directory dir begins with prefix.
+func dirHasPrefix(dir, prefix string) bool {
+	if runtime.GOOS != "windows" {
+		return strings.HasPrefix(dir, prefix)
+	}
+	return len(dir) >= len(prefix) && strings.EqualFold(dir[:len(prefix)], prefix)
+}
+
 // ResolveLocal resolves dependencies for the current project.
 //
 // This begins with the project, builds up a list of external dependencies.
@@ -280,7 +289,7 @@ func (r *Resolver) ResolveLocal(deep bool) ([]string, error) {
 			case LocUnknown, LocVendor:
 				l.PushBack(filepath.Join(r.VendorDir, filepath.FromSlash(imp))) // Do we need a path on this?
 			case LocGopath:
-				if !strings.HasPrefix(info.Path, r.basedir) {
+				if !dirHasPrefix(info.Path, r.basedir) {
 					// FIXME: This is a package outside of the project we're
 					// scanning. It should really be on vendor. But we don't
 					// want it to reference GOPATH. We want it to be detected

